### ISSUE TITLE
takes/src/main/java/org/takes/facets/auth/social/PsLinkedin.java token verify

### ISSUE THREAD

**Issue by Akvadevka** [yegor256/takes#1366]
@yegor256 can you check, please
Here the token is used directly to retrieve user information without verifying its validity or expiration. If the token is expired or revoked, the code will simply throw an error without attempting to refresh the token or handle the situation in any way.
https://github.com/yegor256/takes/blob/2f7bc6ca2810b25b1f475eb410722085592e9c63/src/main/java/org/takes/facets/auth/social/PsLinkedin.java#L138

**Comment by yegor256** [comment]
@Akvadevka you are right, but what can we do in this case? We simply fail with an exception and it's up to the user what to do: relogin again or just stop.

**Comment by Akvadevka** [comment]
@yegor256 You can refresh the token. If refreshing is not possible, explicitly inform the user about the need to re-login instead of throwing an unhandled exception

**Comment by yegor256** [comment]
@Akvadevka we can't refresh inside the framework. It's only the web app can do, but catching the exception and then making a new request to LinkedIn server. Takes is a framework, not an app by itself. We can't be so pro-active in what we do.

**Comment by Akvadevka** [comment]
Ok, i see
